app/src/main/java/org/mozilla/tv/firefox/FirefoxApplication.kt

```diff
@@ -50,6 +51,17 @@ open class FirefoxApplication : LocaleAwareApplication() {
             // Enable crash reporting. Don't add anything above here because if it crashes, we won't know.
             SentryIntegration.init(this, serviceLocator.settingsRepo)
 
+            // Set up Glean by connecting it to the preference flag for data collection and calling
+            // the initialize() function.
+            with(serviceLocator.settingsRepo) {
+                dataCollectionEnabled.observeForever { collectionEnabled ->
```

Comment:
I don't know this API, so somebody from the firefox-tv team should tell us whether or not `setUploadEnabled` gets called at least once here, or only after the settings are modified. Let's get this sorted out before merging.

Reply:
This is how they hook into it for the existing telemetry, I mimic'd that.  I can add an explicit set, I agree that we should explicitly set it here.

Reply:
I think it's fine to just double check with the reviewer that this works the way we expect :)

Reply:
Actually, upon further investigation, this always gets hit on the first pass so the explicit getting of the value shouldn't be needed.

Reply:
Sorry guys, it's been a busy week.

This will be provided a value whenever `SettingsRepo` is instantiated, so it's safe to rely on here.